### ISSUE TITLE
Unauthorized for registry.redhat.io

### ISSUE THREAD

**Issue by michelmeeuwissen** [code-ready/crc#1030]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V 
  * Did you run `crc setup` before starting it Yes?

## CRC version
```bash
crc version: 1.6.0+8ef676f
OpenShift version: 4.3.0 (embedded in binary)
```
  
## CRC status
```bash
CRC VM:          Running
OpenShift:       Running (v4.3.0)
Disk Usage:      11.51GB of 32.72GB (Inside the CRC VM)
Cache Usage:     12.69GB
Cache Directory: C:\Users\***\.crc\cache
```

## CRC config
```bash
<empty>
```

## Host Operating System
```bash
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.18363 N/A Build 18363
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
```

### Steps to reproduce

```
$ crc setup
$ crc start -n 1.1.1.1

[copy pull secret from https://cloud.redhat.com/openshift/install/crc/installer-provisioned]

$ oc login -u kubeadmin -p 7z6T5-*****-oxaoD-p3xQF https://api.crc.testing:6443

$ oc new-project sandbox
// pull-secret.txt from https://cloud.redhat.com/openshift/install/crc/installer-provisioned
$ oc create secret generic pullsecret --from-file=.dockerconfigjson="C:\Users\****\Downloads\pull-secret.txt" --type=kubernetes.io/dockerconfigjson
$ oc secrets link builder pullsecret --for=pull

$ oc new-build --name ansible-slave --binary --strategy docker
// Dir contains Dockerfile
$ oc start-build ansible-slave --from-dir=. --follow


Uploading directory "." as binary input for the build ...
.
Uploading finished
build.build.openshift.io/ansible-slave-1 started
Receiving source from STDIN as archive ...
Caching blobs under "/var/cache/blobs".

Pulling image registry.redhat.io/openshift4/ose-jenkins-agent-base ...
Warning: Pull failed, retrying in 5s ...
Warning: Pull failed, retrying in 5s ...
Warning: Pull failed, retrying in 5s ...
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: unable to retrieve auth token: invalid username/password: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
```

### Dockerfile
```
FROM registry.connect.redhat.com/openshift4/ose-jenkins-agent-base
# FROM quay.io/openshift/origin-jenkins-agent-base:4.3  --> This works

ENV ANSIBLE_VERSION=2.8.2

ADD ubi.repo /etc/yum.repos.d/ubi.repo

RUN INSTALL_PKGS="rh-python36-python-pip" && \
    DISABLE_REPOS=--disablerepo='rhel-*' && \
    yum $DISABLE_REPOS install -y $INSTALL_PKGS && \
    source scl_source enable rh-python36 && \
    scl enable rh-python36 bash && \
    python3 -m pip install ansible==$ANSIBLE_VERSION paramiko && \
    yum $DISABLE_REPOS clean all -y && \
    rm -rf /var/cache/yum

ADD scl_enable /usr/share/container-scripts/
ENV BASH_ENV=/usr/share/container-scripts/scl_enable \
    ENV=/usr/share/container-scripts/scl_enable \
    PROMPT_COMMAND="./usr/share/container-scripts/scl_enable"

USER 1001
```

### Expected
`image ose-jenkins-agent-base` should be pulled from `registry.redhat.io`. Am I doing something wrong?

### Actual

>  Pull image still failed due to error: unable to retrieve auth token: invalid username/password: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials

**Comment by praveenkumar** [comment]
This is duplicate of  #781 and please check https://github.com/code-ready/crc/issues/781#issuecomment-547760709 how to make it work.

**Comment by michelmeeuwissen** [comment]
I think it is not. See 'Steps to reproduce' in the original post.
I actually do :

`oc create secret generic pullsecret --from-file=.dockerconfigjson="C:\Users\****\Downloads\pull-secret.txt" --type=kubernetes.io/dockerconfigjson`

Do I miss something?

**Comment by praveenkumar** [comment]
@michelmeeuwissen in that case you need to follow https://docs.openshift.com/container-platform/4.3/builds/creating-build-inputs.html#builds-docker-credentials-private-registries_creating-build-inputs one. Also this is not CRC specific issue.

After creating the secret you need to do following
```
$ oc new-build --name ansible-slave --binary --strategy docker
$ oc set build-secret --pull bc/ansible-slave pullsecret
$ oc start-build ansible-slave --from-dir=. --follow
Uploading directory "." as binary input for the build ...
..
Uploading finished
build.build.openshift.io/ansible-slave-2 started
Receiving source from STDIN as archive ...
Caching blobs under "/var/cache/blobs".

Pulling image registry.redhat.io/openshift4/ose-jenkins-agent-base ...
Getting image source signatures
Copying blob sha256:0c94cc04e98661c5f25e393a6692ed462109358a8e6aacdeaf51f37e8770bf0e
Copying blob sha256:6adf116da65242d6ddb911d577a85e125a860878301d774703935face61bb4cf
Copying blob sha256:455ea8ab06218495bbbc
```

**Comment by praveenkumar** [comment]
@michelmeeuwissen let us know if https://github.com/code-ready/crc/issues/1030#issuecomment-589508542 solve your issue and we can close this bug?

**Comment by michelmeeuwissen** [comment]
Great ... Thanks!!! 

Here is what I did to make it work:

```
oc new-build --name ansible-slave --binary --strategy docker
oc start-build ansible-slave --from-dir=. --follow
docker login registry.redhat.io -u <LOGIN> -p <PASSWORD>
oc create secret generic redhat-registry --from-file=.dockerconfigjson=/home/****/.docker/config.json --type=kubernetes.io/dockerconfigjson
oc secrets link builder redhat-registry
oc set build-secret --pull bc/ansible-slave redhat-registry
oc start-build ansible-slave --from-dir=. --follow
```


Results in:
```
Uploading directory "." as binary input for the build ...
..
Uploading finished
build.build.openshift.io/ansible-slave-2 started
Receiving source from STDIN as archive ...
Caching blobs under "/var/cache/blobs".

Pulling image registry.redhat.io/openshift4/ose-jenkins-agent-base ...
Getting image source signatures
Copying blob sha256:2a69f56e08208bf952d5307e8b47f8aa4642ff1e5102675330909c99f033827d
Copying blob sha256:bb13d92caffa705f32b8a7f9f661e07ddede310c6ccfa78fb53a49539740e29b
Copying blob sha256:455ea8ab06218495bbbcb14b750a0d644897b24f8c5dcf9e8698e27882583412
Copying blob sha256:e1f7ef81f6300ddeb927955e07d9dfd51b768729fb1630a881bc0bd9418a4f9b
Copying blob sha256:edf561fefe181b648d6502acfd17fd6a6bb8d378057dae272f36ffc850f65bf8
Copying blob sha256:13be0effd90b69b59a7109cbb6133186be48c386dfb5a69e99965ad383451789
Copying config sha256:01a612ddb4ba8fa10b4914cd1c276c6c9e29824264936c66e90d7f240075651f
Writing manifest to image destination
```
